lib/os/heap.c

```diff
@@ -157,14 +157,19 @@ static void free_chunks(struct z_heap *h, chunkid_t c)
 	free_list_add(h, c);
 }
 
+static chunkid_t mem_to_chunkid(struct z_heap *h, void *p)
+{
+	uint8_t *mem = p, *base = (uint8_t *)chunk_buf(h);
+	return (mem - chunk_header_bytes(h) - base) / CHUNK_UNIT;
```

Comment:
Checkpatch is going to yell at you here about a missing blank line after declarations.  I hate that too, but rules are rules.